src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -2574,6 +2575,20 @@ def _validate_vm_vmss_update_ephemeral_placement(cmd, namespace):  # pylint: dis
                                      '--vm-sku is specified.')
 
 
+def _validate_vmss_hyper_v_generation(cmd, namespace):
+    if namespace.security_type is not None:
+        client = _compute_client_factory(cmd.cli_ctx)
+        vmss = client.virtual_machine_scale_sets.get(namespace.resource_group_name, namespace.name)
```

Comment:
This additional REST query will increase the execution time of this command. Is there a way for us not to add the additional request?

Reply:
> This additional REST query will increase the execution time of this command. Is there a way for us not to add the additional request?

We can move the 'validate' logic to `get_vmss_modified` method since `vmss update` is a GenericUpdate command. But I concerned if it is appropriate to  have validation logic in a getter method?